### DISCUSSION TITLE
Safe for dual-boot?

### DISCUSSION THREAD

**Discussion by Bougakov** [openzfsonwindows/openzfs#306] [Q&A]
Hello,

I have a small NAS based on an i3 processor, equipped with two large 3Tb HDDs for storage (system and a ZFS cache partition  resides on a boot SSD). The system dual-boots Ubuntu (default) and Windows 10 (in case some specific Windows app is occasionally needed). 

My question is: is it safe for general home media storage scenario to use a ZFS pool on HDDs *from both Windows and Ubuntu?* It will solve a lot of headaches if I could access files on ZFS while on windows.  Are there any limitations that I should be aware of? Thanks!

P.S. I currently have a second backup of the files stored on the NAS, so I am capable to experiment with my setup.
P.P.S. I am aware that I could run Windows on Ubuntu from a VM, but since I don't need the NAS to be available 24/7 simply booting other OS seems to be a more straightforward approach without an overhead.

**Comment by lundman** [comment]
In theory, it should work but wait for the next rc at least. There are issues using cache-devices, in that we pretty up the name, then Linux can't find it afterwards.

Some care needed when you create the pool to ensure you don't enable features the other OS's version lack.

It was always a goal to be able to import on either, but whether we are there yet I don't know. As you say, you can experiment and see if it performs as you like.
